dotnet/src/SemanticKernel.Skills/Skills.Compression/FileCompressionSkill.cs

```diff
@@ -0,0 +1,125 @@
+    private readonly ILogger<FileCompressionSkill> _logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="FileCompressionSkill"/> class.
+    /// </summary>
+    /// <param name="fileCompressor">File compressor implementation</param>
+    /// <param name="logger">Optional logger</param>
+    public FileCompressionSkill(IFileCompressor fileCompressor, ILogger<FileCompressionSkill>? logger = null)
+    {
+        this._fileCompressor = fileCompressor ?? throw new ArgumentNullException(nameof(fileCompressor));
+        this._logger = logger ?? new NullLogger<FileCompressionSkill>();
+    }
+
+    [SKFunction("Compresses an input file to an output file")]
+    [SKFunctionInput(Description = "Path of file to compress")]
+    [SKFunctionContextParameter(Name = Parameters.DestinationFilePath, Description = "Path of compressed file to create")]
+    public Task CompressFileAsync(string sourceFilePath, SKContext context)
+    {
+        this._logger.LogDebug($"{nameof(CompressFileAsync)} got called");
+
+        if (!context.Variables.Get(Parameters.DestinationFilePath, out string destinationFilePath))
+        {
+            this._logger.LogError($"Missing context variable in {nameof(CompressFileAsync)}");
+            string errorMessage = $"Missing variable {Parameters.DestinationFilePath}";
+            context.Fail(errorMessage);
+
+            return Task.FromException(new KeyNotFoundException(errorMessage));
+        }
+
+        return this._fileCompressor.CompressFileAsync(sourceFilePath, destinationFilePath, context.CancellationToken);
+    }
+
+    [SKFunction("Compresses a directory to an output file")]
+    [SKFunctionInput(Description = "Path of directory to compress")]
+    [SKFunctionContextParameter(Name = Parameters.DestinationFilePath, Description = "Path of compressed file to create")]
+    public Task CompressDirectoryAsync(string sourceDirectoryPath, SKContext context)
+    {
+        this._logger.LogDebug($"{nameof(CompressDirectoryAsync)} got called");
+
+        if (!context.Variables.Get(Parameters.DestinationFilePath, out string destinationFilePath))
+        {
+            this._logger.LogError($"Missing context variable in {nameof(CompressDirectoryAsync)}");
+            string errorMessage = $"Missing variable {Parameters.DestinationFilePath}";
+            context.Fail(errorMessage);
+
+            return Task.FromException(new KeyNotFoundException(errorMessage));
+        }
+
+        return this._fileCompressor.CompressDirectoryAsync(sourceDirectoryPath, destinationFilePath, context.CancellationToken);
+    }
+
+    [SKFunction("Decompresses an input file")]
+    [SKFunctionInput(Description = "Path of file to decompress")]
+    [SKFunctionContextParameter(Name = Parameters.DestinationDirectoryPath, Description = "Directory into which to extract the decompressed content")]
+    public Task DecompressFileAsync(string sourceFilePath, SKContext context)
```

Comment:
this is a bit worrying. How does it handle file overwrite? what if I unzip over critical user files like config, apps, etc?

Reply:
This fails on attempted overwrites with
Exception thrown: 'System.IO.IOException' in System.Private.CoreLib.dll
An exception of type 'System.IO.IOException' occurred in System.Private.CoreLib.dll but was not handled in user code
The file 'C:\Users\gillahaye\AppData\Local\Temp\biorgmb1.ojd.zip' already exists.

If we go zero-trust with regards to file extraction, I guess we could only extract to a "sandbox" folder (and subfolders).  Are you suggesting something of this nature?

Reply:
considering it's a library, there will be very different developers with different expectations, many not even thinking about this. I would make sure the code is "safe" -- deferring to you and other reviewers to agree what that mean exactly

Reply:
We feel as the group working on Sample 4 that simply not overwriting existing files is sufficiently "safe" for now.